README: Mention which CIs support ARM builds (#445)
joerick/cibuildwheel: README.md

`cibuildwheel` currently works on **Travis CI**, **Azure Pipelines**, **AppVeyor**, **GitHub Actions**, **CircleCI**, and **Gitlab CI**. Check the table below for supported platforms on each service:

> |                 | Linux | macOS | Windows | Supports ARM |

Reviewer: Linux ARM

Reviewer 2: ```suggestion
|                 | Linux | macOS | Windows | Linux ARM |
```

|-----------------|-------|-------|---------|--------------|
| Azure Pipelines | ✅    | ✅    | ✅      | ✴️¹           |
| Travis CI       | ✅    | ✅    | ✅      | ✅           |
| AppVeyor        | ✅    | ✅    | ✅      |              |

> | GitHub Actions  | ✅    | ✅    | ✅      |              |

Reviewer: also support self hosted ARM https://github.blog/changelog/2019-12-03-github-actions-self-hosted-runners-on-arm-architectures/

But We have no possibilities to test if cibuilwheel works properly on such machines.

Reviewer 2: ```suggestion
| GitHub Actions  | ✅    | ✅    | ✅      | ✴️¹           |
```

Reviewer 2: Ahh, I just added this, assuming it would work, but maybe we should not if we can't test it? I guess Azure was tested at one point by someone? Though I really think it would work on GHA if it works on Azure, they are basically the same. And "third-party" already indicates a lot of work needs to be done to set it up.

Author: I suppose we could dial back the footnote to explain the situation: "These CIs require self-hosted runners for ARM. cibuildwheel probably works but this has not been confirmed."

| CircleCI        | ✅    | ✅    |         |              |
| Gitlab CI       | ✅    |       |         |              |

> <sup>¹ Supported with a "third-party build host".</sup><br>

Reviewer 2: ```suggestion
<sup>¹  Requires a "third-party build host"; expected to work with cibuildwheel but not directly tested by our CI.</sup><br>
```
